controller/instance_manager_controller.go

```diff
@@ -585,14 +585,24 @@ func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.I
 		}
 
 		hasHealthyReplicaOnAnotherNode := false
+		isUnusedReplicaOnCurrentNode := false
 		for _, r := range replicas {
 			if r.Spec.HealthyAt != "" && r.Spec.FailedAt == "" && r.Spec.NodeID != imc.controllerID {
 				hasHealthyReplicaOnAnotherNode = true
 				break
 			}
+			// If a replica has never been started, there is no data stored in this replica, and
+			// retaining it makes no sense for HA.
+			// Hence Longhorn doesn't need to block the PDB removal for the replica.
+			// This case typically happens on a newly created volume that hasn't been attached to any node.
+			// https://github.com/longhorn/longhorn/issues/2673
+			if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" && r.Spec.NodeID == imc.controllerID {
+				isUnusedReplicaOnCurrentNode = true
+				break
+			}
 		}
 
-		if !hasHealthyReplicaOnAnotherNode {
+		if !hasHealthyReplicaOnAnotherNode && !isUnusedReplicaOnCurrentNode {
```

Comment:
Should we check `isUnusedReplicaOnCurrentNode` for all volumes' replicas instead of at least one volume meeting this condition?

cc @shuo-wu

Reply:
When I deploy a volume with 3 replicas, all three nodes will encounter `pod disruption budgets`.

Reply:
Just check again, LGTM now